Autonomous Integration

StephenWelch opened the pull request.

Commit:
Add support for yaw/pitch/roll rates

Commit:
Add inches to ticks conversion

Commit:
Add drive straight implementation

Commit:
Reset Pigeon IMU config to factory default

Commit:
Add unit test shell

Commit:
Rename velocity mode to motion magic

Commit:
Fix tests

Commit:
Simplify heading control and unit tests

StephenWelch committed:
Refactor TargetLock

GlebBabahov committed:
Remade TrackingType enum and some logic

needs to be tested

GlebBabahov committed:
Clean ups

GlebBabahov committed:
wip

StephenWelch committed:
Working tracking

GlebBabahov committed:
Tweaks

GlebBabahov committed:
Merge remote-tracking branch 'origin/experimental/vision-tracking' into experimental/vision-tracking

GlebBabahov committed:
Merge remote-tracking branch 'origin/master' into experimental/vision-tracking

GlebBabahov committed:
Finalizing targetlock

StephenWelch committed:
Clarify javadocs

StephenWelch committed:
Update superstructure state machine

- Set elevator to a "handoff height"
- Elevator cannot be moved when intaking/handing off

StephenWelch committed:
Remove unit tests

StephenWelch committed:
Add setter for elevator position is superstructure

StephenWelch committed:
Fix unit test

StephenWelch committed:
Merge and resolve conflicts

StephenWelch committed:
Merge remote-tracking branch 'origin/master' into experimental/superstructure

StephenWelch committed:
Refactor ICommand to ACommand

StephenWelch committed:
Add InstantCommand

Takes in a Runnable so a lambda can be passed in

StephenWelch committed:
Add Empty Command

StephenWelch committed:
Add basic module commands

StephenWelch committed:
Add handoff command

StephenWelch committed:
Add functional command

StephenWelch committed:
Add hatch extension

StephenWelch committed:
Merge branch 'hatch-extension' into experimental/superstructure

StephenWelch committed:
Add hatch grab command

StephenWelch committed:
Add hatch release command

StephenWelch committed:
Add cargo intake command

StephenWelch committed:
Add set intake position command

StephenWelch committed:
Add hatch grabber extend/retract command

StephenWelch committed:
Rename elevator position command

StephenWelch committed:
Add hatch handoff command

StephenWelch committed:
Add intake commands

StephenWelch committed:
Add new elevator setpoints

StephenWelch committed:
Add setOuttaking() to intake

StephenWelch committed:
Add new driver controls

StephenWelch committed:
Fix test

StephenWelch committed:
Remove unnecessary stuff from superstructure

StephenWelch committed:
Compile fixes

StephenWelch committed:
Merge remote-tracking branch 'origin/experimental/auto/drivestraight'

StephenWelch committed:
Add autonomous routine class and system for managing trajectory generation

oconnelc committed:
Merge branch 'master' into experimental/superstructure

StephenWelch committed:
Construct autonomous routines when AutonomousRoutines is instantiated

StephenWelch committed:
Change heading controller input range

StephenWelch committed:
Reduce DriveStraight speed

StephenWelch committed:
Start loops before modules

StephenWelch committed:
Remove unnecessary call

StephenWelch committed:
Use CommandManager to manage autonomous and teleop commands

StephenWelch committed:
Revert dumb change

StephenWelch committed:
Revert dumb change

StephenWelch committed:
Merge remote-tracking branch 'origin/experimental/superstructure' into experimental/superstructure

StephenWelch committed:
Remove inversion on NavX heading

StephenWelch committed:
Fix Drivestraight

- Clamp linear output by fixed amount
- Reduce heading gain (mistake)
- Add hard clamp on linear output

StephenWelch committed:
Adjust field dimensions

StephenWelch committed:
Set reversed flag on middle left hatch to loading station path

StephenWelch committed:
Adjust drive forward distance

StephenWelch committed:
Post diagnostics information to NetworkTables in FollowTrajectory

StephenWelch committed:
Merge integration/auto

StephenWelch committed:
Merge remote-tracking branch 'origin/master' into experimental/vision-tracking

kc5tjn commented on common/src/main/java/us/ilite/common/lib/util/Conversions.java:
```diff
@@ -32,6 +32,10 @@ public static double ticksToInches(int ticks) {
         return ticksToRotations(ticks) * SystemSettings.kDriveWheelCircumference;
     }
 
+    public static int inchesToTicks(double inches) {
```
Seems like a very generic name for a very specific conversion, or is this conversion applicable for more than the drive train?

kc5tjn commented on robot/src/main/java/us/ilite/robot/Robot.java:
```diff
@@ -49,22 +41,22 @@
     private HatchFlower mHatchFlower = new HatchFlower();
 
-    private DriverInput mDriverInput = new DriverInput(mDrive, mElevator, mHatchFlower, mSuperstructure, mData);
+    private DriverInput mDriverInput = new DriverInput(mDrive, mElevator, mHatchFlower, mAutonomousCommandManager, mTeleopCommandManager, mData);
```
Be careful with this merge, I've seen inputs to the constructor dropped in earlier merges.

kc5tjn commented on robot/src/main/java/us/ilite/robot/Robot.java:
```diff
@@ -114,16 +102,16 @@ public void autonomousInit() {
+
         // Init modules after commands are set
-        mRunningModules.setModules(mSuperstructure);
+        mRunningModules.setModules(mAutonomousCommandManager, mTeleopCommandManager);
```
Was mSuperstructure removed for testing?

kc5tjn commented on robot/src/main/java/us/ilite/robot/auto/AutonomousRoutines.java:
```diff
@@ -0,0 +1,57 @@
+        mMiddleToMiddleCargoToSideRocket = new MiddleToMiddleCargoToSideRocket(mTrajectoryGenerator);
+    }
+
+    public void generateTrajectories() {
```
Seems to be a very generic name for very specific trajectories

StephenWelch committed:
Merge branch 'master' into experimental/vision-tracking

StephenWelch committed:
Merge branch 'master' into experimental/superstructure

StephenWelch committed:
Merge branch 'master' into integration/auto

StephenWelch committed:
Merge branch 'experimental/vision-tracking' into integration/auto

StephenWelch committed:
Fix merge conflicts

StephenWelch committed:
Invert NavX heading for odometry

StephenWelch committed:
Mirror all generated trajectories

Makes sure that (0, 0) is in top left, meaning that we can give the robot trajectories with an upper-left hand origin and have things work fine

StephenWelch committed:
Add flip parameter to TrajectoryGenerator

StephenWelch committed:
Move autonomous sequences to individual path class

StephenWelch committed:
Exit TargetLock when we acquire then lose target

StephenWelch committed:
Run command managers in teleop

StephenWelch committed:
Limit max turn power

StephenWelch committed:
Tune target lock PID

StephenWelch committed:
Make minor correction to command manager logging

StephenWelch committed:
Update target tracking pipelines

StephenWelch committed:
Merge origin/master

StephenWelch committed:
Merge popped stash

StephenWelch committed:
Fix compile error

StephenWelch committed:
Add FunctionalCommand from experimental/superstructure

StephenWelch committed:
Cleanup and testing

StephenWelch committed:
Clean up TargetLock

StephenWelch committed:
Add wrapper command for limelight

StephenWelch committed:
Fix codex mapping in Limelight

- Log data to codex in periodicInput()
- Only check for visionTarget validity when needed

StephenWelch committed:
WIP Test

StephenWelch committed:
Revert DriveHardware changes

This reverts changes from commit b35b990249abba57998b6fc434e05e9824f8141a.

StephenWelch committed:
Remove drive shifting

StephenWelch committed:
Merge integration/master

StephenWelch committed:
Fix test compile errors

StephenWelch committed:
Remove print

StephenWelch committed:
Merge remote-tracking branch 'origin/master' into integration/auto
